src/rgw/driver/rados/rgw_tools.cc

```diff
@@ -422,3 +433,62 @@ void rgw_complete_aio_completion(librados::AioCompletion* c, int r) {
   librados::CB_AioCompleteAndSafe cb(pc);
   cb(r);
 }
+
+void rgw_fix_etag(CephContext *cct, bufferlist& etagbl)
```

Comment:
Maybe you would consider a function comment to explain what's going on in here since it's non-obvious?